src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,102 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] array;
+    private int arraySize;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_SIZE];
+        arraySize = 0;
+    }
+
+    private Object[] grow() {
+        return Arrays.copyOf(array, array.length + (array.length >> 1));
+    }
 
     @Override
     public void add(T value) {
-
+        if (arraySize == array.length) {
+            array = grow();
+            array[arraySize] = value;
+            arraySize++;
+        } else {
+            array[arraySize] = value;
+            arraySize++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
let’s also check the index - it should not be less than 0 or greater than size (it can be equal to size though)